crates/query-engine/translation/src/translation/query/root.rs

```diff
@@ -249,6 +252,8 @@ pub fn translate_rows_query(
                             current_table.reference.clone(),
                         )?;
                     }
+                    // Native mutations will not have limit or offset
+                    CollectionInfo::NativeMutation { .. } => {}
```

Comment:
If this condition matches, since this is inside the if condition for `has_limit_or_offset`, should we be throwing an error here letting the user know that is not permitted?

Reply:
Makes sense. Although, we will never match this case, to be on the safer side I will throw an internal error that something unexpected happened.